source/common/http/async_client_impl.cc

```diff
@@ -79,6 +79,9 @@ void AsyncStreamImpl::encodeHeaders(HeaderMapPtr&& headers, bool end_stream) {
       },
       nullptr);
 #endif
+  if (remote_closed_) {
```

Comment:
How can this one happen? I can see how data can happen. I think trailers can also happen. We need to guard against that also I think.